Source/WebCore/html/canvas/CanvasRenderingContext2DBase.cpp

```diff
@@ -2222,7 +2249,36 @@ ExceptionOr<Ref<ImageData>> CanvasRenderingContext2DBase::getImageData(int sx, i
 
     ASSERT(pixelBuffer->format().colorSpace == toDestinationColorSpace(computedColorSpace));
 
-    return { { ImageData::create(static_reference_cast<ByteArrayPixelBuffer>(pixelBuffer.releaseNonNull())) } };
+    auto imageData = ImageData::create(static_reference_cast<ByteArrayPixelBuffer>(pixelBuffer.releaseNonNull()));
+    
+#if ENABLE(NOISY_APIS)
+    // auto identifier = IdentifierProvider::provider().identifier();
+    // (void)identifier;
+    // FIXME: Use identifier...
+
+    auto& data = imageData->pixelBuffer()->data();
+    // WTFLogAlways("getImageData: STARTING");
+    // WTFReportBacktrace();
+    for (size_t i = 0; i < data.length(); i += 4) {
+        if (randomNumber() > 0.75) {
+            int fivePercentOffset = static_cast<int>(randomNumber() * 26 - 13);
+            // WTFLogAlways("Random offset: %d", five_percent_offset);
+            size_t channel = static_cast<size_t>(randomNumber() * 3);
```

Comment:
Do you intend to potentially select the alpha channel? If so, this should be `auto channel = static_cast<size_t>(randomNumber() * 4);`

Reply:
The alpha channel is excluded on purpose.